stablehlo/transforms/StablehloRefineShapes.cpp

```diff
@@ -1089,18 +1089,30 @@ struct StablehloRefineShapesPass
     // Current use cases are served well by inlining multiple functions into
     // a single function, so we leave native support for multiple functions to
     // future work.
+    // To enable modules that contain CustomCallOp::called_computations,
+    // we allow multiple functions, in which case we only refine the main
+    // function called "main", assuming that the called computations will have
+    // static shapes. Lifting this assumption and expanding refinement to
+    // multiple functions is left for future work.
     ModuleOp module = getOperation();
     auto funcs = llvm::to_vector(module.getOps<func::FuncOp>());
     if (funcs.empty()) return;
-    if (funcs.size() != 1) {
-      module.emitOpError() << "must have no more than one function";
+    func::FuncOp func;
+    if (funcs.size() == 1) {
```

Comment:
Is there no standard name for a single function? Any way to restrict this to only refine `@main`?

Reply:
Will be addressed in https://github.com/openxla/stablehlo/issues/1494. For now, merging this PR to simplify integrate.